resources/telemetry.md

```diff
@@ -0,0 +1,521 @@
+---
+layout: default
+nav_order: 3
+parent: Resources
+grand_parent: Maintainer Docs
+title: Telemetry
+---
+
+# {{ page.title }}
+{:.no_toc}
+
+## Overview
+{:.no_toc}
+
+The RAPIDS team collects build-time and test-time telemetry from our CI jobs.
+These data are used to provide insights into potential speed impacts and
+optimizations that might be worthwhile.
+
+### Intended audience
+{: .no_toc }
+
+Operations
+{: .label .label-purple}
+
+## Table of contents
+{: .no_toc .text-delta }
+
+1. TOC
+{:toc}
+
+## Introduction
+
+This is an overview of how telemetry data is collected, how it is transmitted and recorded, and how visualizations can be obtained using Grafana.
+
+1. **Setup Phase**: `telemetry-setup` job runs first, creating base environment variables
+2. **Execution Phase**: Reusable workflows load variables, add metadata, and execute build/test operations. They stash results to be associated with spans later.
+3. **Collection Phase**: `telemetry-summarize` job parses GitHub Actions metadata and associates it with workflow attributes
+4. **Transmission Phase**: Data sent via OpenTelemetry SDK through Vector to Tempo. Also happens in telemetry-summarize job in pipeline.
+5. **Analysis Phase**: Grafana queries Tempo data using TraceQL for visualization and analysis
+
+![](/assets/images/telemetry/mermaid-workflow.png)
+
+<!-- NOTE: For source code of mermaid diagram, see assets/images/telemetry/mermaid-workflow.md  -->
+
+## Code changes for projects to add telemetry
+
+These are the general steps necessary to add telemetry to a repository that does
+not currently have it. This does not describe changes to shared-workflows as
+implementation details. In the top-level workflow, such as
+`cudf/.github/workflows/pr.yaml`:
+
+* Add a job for telemetry-setup, and add that job name to the pr-builder `needs` collection.
+
+{% raw %}
+```
+jobs:
+  # Please keep pr-builder as the top job here
+  pr-builder:
+    needs:
+      - check-nightly-ci
+      - changed-files
+
+      ... <other job names> ...
+
+      - telemetry-setup
+  telemetry-setup:
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    env:
+      OTEL_SERVICE_NAME: 'pr-cudf'
```

Comment:
Does this service name have a specific meaning? Does it need to match the repo name, or match some other internal place with the same name?